util/helm/creds.go

```diff
@@ -146,8 +146,7 @@ func (creds AzureWorkloadIdentityCreds) GetAccessToken() (string, error) {
 		return "", fmt.Errorf("failed to get Azure access token after challenge: %w", err)
 	}
 
-	// Access token has a lifetime of 3 hours
-	storeAzureToken(key, token, 2*time.Hour)
+	storeAzureToken(key, token, 10*time.Minute)
```

Comment:
+1. The token should be a JWT, and I assume you have the expiration claim set. You can use the expiration claims (minus 5 minutes just to be safe) and use that as the expiration for the cache instead of an hardcoded value.

Reply:
Done